great_expectations/rule_based_profiler/attributed_resolved_metrics.py

```diff
@@ -107,17 +89,9 @@ def _detect_illegal_array_type_or_shape(values: MetricValues) -> bool:
         return False
 
     if _detect_illegal_array_type_or_shape(values=metric_values):
-        value: MetricValue
-        if (
-            metric_values is not None
-            and isinstance(metric_values, (list, tuple))
-            and all(value is None for value in metric_values)
-        ):
-            return np.asarray(metric_values)
-
         return metric_values
-    else:
-        return np.asarray(metric_values)
+
+    return np.asarray(metric_values)
```

Comment:
`np.asarray(metric_values)` is not a `MetricValues` which is:
`MetricValues = Union[
Union[
    MetricValue, List[MetricValue], Set[MetricValue], Tuple[MetricValue, ...]
]`
The function return types should be updated.

Reply:
@billdirks Looking.

Reply:
@billdirks Per our conversation -- it is included.  Thanks.

Reply:
I see, `MetricValue` has collections in it which I did not expect (and `Any`!).